README.md

```diff
@@ -12,219 +12,234 @@
-    - [aNMM](#anmm)
-    - [DUET](#duet)
-    - [MatchPyramid](#matchpyramid)
-    - [match-LSTM](#match-lstm)
-    - [DRMM-TKS](#drmm-tks)
-    - [BiMPM](#bimpm)
-    - [DeepRank](#deeprank)
-    - [Match-SRNN](#match-srnn)
-    - [Co-PACRR](#co-pacrr)
-    - [DLCM](#dlcm)
-    - [Siamese-LSTM](#siamese-lstm)
-    - [ESIM](#esim)
-    - [MCAN](#mcan)
-    - [MIX](#mix)
-    - [HiNT](#hint)
+Text matching is a core component in many natural language processing tasks, where many task can be viewed as a matching between two texts input.
 
+​							$$\text{Match}(s, t) = g(f(\psi(s), \phi(t)))$$
 
-### Models
-------
+Where $s$ and $t$ are source text input and target text input, respectively. The $\psi$ and $\phi$ are representation function for input $s$ and $t$, respectively. The $f$ is the interaction function, and $g$ is the aggregation function. The representative matching tasks are as follows:
 
-#### GLM:
-![status](artworks/not-in-plan.svg)
+<table>
+  <tr>
+    <th width=30%, bgcolor=#999999 >Tasks</th> 
+    <th width=20%, bgcolor=#999999>Source Text</th>
+    <th width="20%", bgcolor=#999999>Target Text</th>
+  </tr>
+  <tr>
+    <td align="center", bgcolor=#eeeeee> Ad-hoc Information Retrieval </td>
+    <td align="center", bgcolor=#eeeeee> query </td>
+    <td align="center", bgcolor=#eeeeee> document (title/content) </td>
+  </tr>
+    <tr>
+    <td align="center", bgcolor=#eeeeee> Community Question Answer </td>
+    <td align="center", bgcolor=#eeeeee> question </td>
+    <td align="center", bgcolor=#eeeeee> question/answer </td>
+  </tr>
+  <tr>
+    <td align="center", bgcolor=#eeeeee> Paraphrase Indentification </td>
+    <td align="center", bgcolor=#eeeeee> string 1 </td>
+    <td align="center", bgcolor=#eeeeee> string 2 </td>
+  </tr>
+  <tr>
+    <td align="center", bgcolor=#eeeeee> Natural Language Inference </td>
+    <td align="center", bgcolor=#eeeeee> premise </td>
+    <td align="center", bgcolor=#eeeeee> hypothesis </td>
+  </tr>
+</table>
 
-A Word Embedding based Generalized Language Model for Information Retrieval.
-[website](https://dl.acm.org/citation.cfm?id=2767780) | [paper](http://www.cs.unibo.it/~montesi/CBD/Articoli/2015_A%20Word%20Embedding%20based%20Generalized%20Language%20Model%20for%20Information%20Retrieval.pdf)
 
-#### DESM:
-![status](artworks/not-in-plan.svg)
 
-A Dual Embedding Space Model for Document Ranking.
-[website](https://arxiv.org/abs/1602.01137) | [paper](https://arxiv.org/pdf/1602.01137.pdf)
+### Ad-hoc Information Retrieval
 
+---
 
-#### DSSM:
-![status](artworks/ready.svg)
+**Information retrieval** (**IR**) is the activity of obtaining information system resources relevant to an information need from a collection. Searches can be based on full-text or other content-based indexing.  Here, the **Ad-hoc information retrieval** refer in particular to text-based retrieval where the number of queries is huge. Some benchmark datasets are listed in the following,
```

Comment:
The term ad-hoc refers to the scenario where documents in the collection remain relatively static while new queries are submitted to the system continually. (cited from the [survey](https://arxiv.org/pdf/1903.06902.pdf) )